providers/storage/storageDocStore.js

```diff
@@ -47,43 +47,36 @@ class AzureStorageDocStore {
       blobMetadata.extra = JSON.stringify(document._metadata.extra);
     }
     const options = { metadata: blobMetadata, contentSettings: { contentType: 'application/json' } };
-    this.service.createBlockBlobFromText(this.name, blobName, text, options, (error, result, response) => {
-      if (error) {
+    const dataStream = new Readable();
+    dataStream.push(JSON.stringify(document));
+    dataStream.push(null);
+    dataStream
+      .pipe(this.service.createWriteStreamToBlockBlob(this.name, blobName, options))
+      .on('error', (error) => {
         return deferred.reject(error);
-      }
-      memoryCache.put(document._metadata.url, { etag: document._metadata.etag, document: document }, this.options.ttl);
-      deferred.resolve(blobName);
-    });
+      })
+      .on('finish', () => {
+        deferred.resolve(blobName);
+      });
     return deferred.promise;
   }
 
   // TODO: Consistency on whether key is a URL or URN
   get(type, key) {
-    const cached = memoryCache.get(key);
-    if (cached) {
-      return Q(cached.document);
-    }
-
     const deferred = Q.defer();
     const blobName = this._getBlobNameFromKey(type, key);
     this.service.getBlobToText(this.name, blobName, (error, text, blob, response) => {
```

Comment:
I've tested both upsert with createWriteStreamToBlockBlob and the current version of get function using getBlobToText and they both work with large files (hundreds of MB).